doc/design/xgboost_feature_column.md

```diff
@@ -105,20 +150,48 @@ class ListedFeatureColumnTransformer(BaseFeatureColumnTransformer):
+would be writen into SVM file, then it can be loaded in the following train step.
 
-Another concern is that we should perform the same data pre-processing in prediction stage. So we should save the feature columns
-of training, so that it can be loaded in prediction stage. Besides, the codegen during prediction stage should also generate the same
-transformation codes as training stage.
+Another concern is that we should perform the same data pre-processing in prediction/evaluation stage. So we should save the feature columns
+of training, so that it can be loaded in prediction/evaluation stage. Besides, the codegen during prediction/evaluation stage should also 
+generate the same transformation codes as training stage.
 
 It should be noticed that `EMBEDDING` is not supported in this design doc. It is because that the `EMBEDDING` feature column may contain
 trainable parameters, and these parameters cannot be updated in XGBoost training process.
 
+## Export the XGBoost models to PMML/Treelite file
+
+XGBoost supports 2 kinds of APIs to train a model:
+
+- `xgboost.train` . We use this API in our current implementation. The returned Booster can be saved to the format that can be loaded
+by Treelite APIs but not by PMML APIs.
+
+- `xgboost.XGBClassifier/XGBRegressor/XGBRanker` . [Sklearn2pmml](https://github.com/jpmml/sklearn2pmml) or 
+[Nyoka](https://github.com/nyoka-pmml/nyoka) can only export models built by these APIs to PMML format. But this APIs may be not very easy to use, because:
+
+    - We must distinguish whether the model is a classifier/regressor/ranker beforehand.
+    - The constructors of `xgboost.XGBClassifier/XGBRegressor/XGBRanker` mix up the Booster parameters and training parameters together. For example,
+      `booster` is one of the Booster parameters, and `n_estimators` is one of the training parameters, but both of them appear in the constructors.
+      This makes us hard to distinguish model parameters and training parameters in SQLFlow codes.
+    - Names of some of the parameters in `xgboost.XGBClassifier/XGBRegressor/XGBRanker` are different from `xgboost.train` . For example, `n_estimators`
+      in `xgboost.XGBClassifier/XGBRegressor/XGBRanker` is the same as `num_boost_round` in `xgboost.train`.
+      
+Therefore, we prefer to use `xgboost.train` API to perform training in the this design, and export PMML/Treelite files in the following ways:
+
+- PMML file can be exported by: 
+    - Call `Booster.load_model()` to load trained model.
+    - Check the Booster objective to build one of `xgboost.XGBClassifier/XGBRegressor/XGBRanker` .
+    - Call `xgboost.XGBClassifier/XGBRegressor/XGBRanker.load_model()` to load the trained model again.
+    - Build a sklearn pipeline using the pre-built `xgboost.XGBClassifier/XGBRegressor/XGBRanker` .
+    - Save the pipeline using [Sklearn2pmml](https://github.com/jpmml/sklearn2pmml) or [Nyoka](https://github.com/nyoka-pmml/nyoka) as a PMML file.
```

Comment:
We are going to use SkLearn2PMML package according to [comment1](https://github.com/sql-machine-learning/sqlflow/issues/2190#issuecomment-631846387), [comment2](https://github.com/sql-machine-learning/sqlflow/issues/2190#issuecomment-632317830) and [here](https://github.com/nyoka-pmml/nyoka/issues/33#issuecomment-626772362), right?

Reply:
Yes, I perfer sklearn2pmml. The only concern is that sklearn2pmml relies on Java 1.8+. PAI has already installed Java 1.8.

Reply:
> Yes, I perfer sklearn2pmml. The only concern is that sklearn2pmml relies on Java 1.8+. PAI has already installed Java 1.8.

So will it be a blocking issue to use sklearn2pmml in PAI?